FFmpegInterop/ExternalSubtitleProvider.h

```diff
@@ -0,0 +1,110 @@
+#pragma once
+#include <pch.h>
+#include "StreamInfo.h"
+#include <mutex>
+#include <FFmpegInteropConfig.h>
+#include <collection.h>
+
+using namespace Platform;
+using namespace Windows::Foundation;
+using namespace Windows::UI::Xaml;
+using namespace Windows::UI::Core;
+
+using namespace std;
+
+namespace FFmpegInterop {
+	ref class ExternalSubtitleProvider
+	{
+
+		DispatcherTimer^ timer;
+		SubtitleStreamInfo^ externalStream;
+
+		recursive_mutex mutex;
+		TimeSpan oldOffset, newOffset;
+	internal:
+
+		property CoreDispatcher^ Dispatcher;
+		property SubtitleStreamInfo^ ExternalStream
+		{
+			SubtitleStreamInfo^ get()
+			{
+				return externalStream;
+			}
+		}
+
+		ExternalSubtitleProvider(CoreDispatcher^ p_dispatcher, SubtitleStreamInfo^ p_externalStream)
+		{
+			this->Dispatcher = p_dispatcher;
+			this->externalStream = p_externalStream;
+		}
+
+		void SetSubtitleOffset(TimeSpan newOffset, TimeSpan currentOffset)
+		{
+			mutex.lock();
+			this->newOffset = newOffset;
+			this->oldOffset = currentOffset;
```

Comment:
I think here is a potential race condition. If set method is called twice before tick comes, then the calculations in OnTick will be wrong (because oldDelay is not correct anymore). It could be fixed like this: Have two fields currentDelay and newDelay. Initialize currentDelay in the constructor, with the value from the config class. In the set method, only set newDelay. Now in OnTick, you will always have the correct currentDelay, plus the latest newDelay.